app/models/test_track/vary_dsl.rb

```diff
@@ -34,7 +34,7 @@ def default(variant, &block)
   delegate :split_name, to: :assignment
 
   def split
-    split_registry && split_registry[split_name]
+    split_registry && split_registry['splits'][split_name] && split_registry['splits'][split_name]['weights']
```

Comment:
This nested hash lookup thing happening in several places is setting off my spideysense. It feels like we’re developing some object envy. 

Like maybe it would be nicer if split_registry responded to weights_for_split(split_name) or some sort of thing like that to provide some encapsulation?

Reply:
wrapped it up into a global SplitRegistry cache.